### PR TITLE
docs: Update E2E test documentation

### PR DESCRIPTION
**Pull request by dcalhoun**
## Related PRs
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6307

## What?

Update E2E test documenation for the Gutenberg for Mobile Apps project.

## Why?

Relates to #55166. Address breaking changes from upgrading to Appium 2. Reduce
verbosity and duplication in documentation to improve approachability and
maintainability.

## How?

Update all relevant documentation.

## Testing Instructions

Review the rendered documentation. Verify all links work as expected.

### Testing Instructions for Keyboard

n/a

## Screenshots or screencast 
n/a

### CODE REVIEW

File: docs/contributors/code/react-native/README.md
Comment (dcalhoun) [inline docs/contributors/code/react-native/README.md:1]: Changes remove outdated information and update links.

File: docs/contributors/code/react-native/getting-started-react-native.md
Comment (dcalhoun) [inline docs/contributors/code/react-native/getting-started-react-native.md:1]: Changes replace duplicative documentation with links to documentation in a single location.

File: packages/react-native-editor/__device-tests__/CONTRIBUTING.md
Comment (dcalhoun) [inline packages/react-native-editor/__device-tests__/CONTRIBUTING.md:1]: Changes replace this with a section in the primary document. The section is far smaller to reduce the maintenance burden for the documentation.

File: packages/react-native-editor/__device-tests__/README.md
Comment (dcalhoun) [inline packages/react-native-editor/__device-tests__/README.md:1]: Update with the latest information to serve as the primary document for E2E testing for the Gutenberg for Mobile Apps project.

**Review by geriux — APPROVED** [review]
LGTM! Nice work updating the docs! I've confirmed the links work as expected. 🚀

**Pull request merged**